cheroot/server.py

```diff
@@ -827,12 +827,15 @@ def read_request_line(self):
             self.simple_response('400 Bad Request', 'Malformed Request-URI')
             return False
 
+        uri_is_absolute_form = (scheme or authority)
+
         if self.method == b'OPTIONS':
             # TODO: cover this branch with tests
             path = (
                 uri
                 # https://tools.ietf.org/html/rfc7230#section-5.3.4
-                if self.proxy_mode or uri == ASTERISK
+                if (self.proxy_mode and uri_is_absolute_form)
+                or uri == ASTERISK
```

Comment:
I'm pretty sure this line can be eliminated as `urllib.parse.urlsplit('*').path == '*'` anyway.

Reply:
Removed in dc9592b6.

Reply:
@jaraco where did you see `urllib.parse.urlsplit('*').path == '*'`?

Reply:
Because I don't see any code checking for `ASTERISK` now.

Reply:
I think that this change is wrong. And the problem with this PR was this hack: https://github.com/cherrypy/cherrypy/blob/535605c/cherrypy/_cpwsgi_server.py#L31

It was temporary until this PR's figured out (https://github.com/cherrypy/cherrypy/commit/c87bc9f73740dfe35e4d2ca9082dccb6278bdc53). We should revert it to `proxy_mode=False` now and see if it's ok.